src/postfork.cpp

```diff
@@ -63,39 +63,56 @@ static void debug_safe_int(int level, const char *format, int val) {
 /// process if it is the first in a JOB_CONTROL job.
 /// Returns true on sucess, false on failiure.
 bool child_set_group(job_t *j, process_t *p) {
-    bool retval = true;
     if (j->get_flag(job_flag_t::JOB_CONTROL)) {
         if (j->pgid == INVALID_PID) {
             j->pgid = p->pid;
         }
-        if (setpgid(p->pid, j->pgid) < 0) {
-            char pid_buff[128];
-            char job_id_buff[128];
-            char getpgid_buff[128];
-            char job_pgid_buff[128];
-            char argv0[64];
-            char command[64];
-
-            format_long_safe(pid_buff, p->pid);
-            format_long_safe(job_id_buff, j->job_id);
-            format_long_safe(getpgid_buff, getpgid(p->pid));
-            format_long_safe(job_pgid_buff, j->pgid);
-            narrow_string_safe(argv0, p->argv0());
-            narrow_string_safe(command, j->command_wcstr());
-
-            debug_safe(
-                1, "Could not send own process %s, '%s' in job %s, '%s' from group %s to group %s",
-                pid_buff, argv0, job_id_buff, command, getpgid_buff, job_pgid_buff);
-
-            safe_perror("setpgid");
-            retval = false;
+
+        while (int result = setpgid(p->pid, j->pgid) != 0) {
```

Comment:
This groups like

```
(int result = setpgid()) != 0
```

and not like

```
int result = (setpgid() != 0)
```

right?

(I can never recall operator precedence so I just add parens everywhere)

Reply:
Good catch.

Reply:
This was actually even more broken than that, I was testing the 0/1 `result` instead of `errno` 🤦‍♂️